cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -291,8 +291,8 @@ func ValidateEtcd(e *kubeadm.Etcd, fldPath *field.Path) field.ErrorList {
 func ValidateCertSANs(altnames []string, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	for _, altname := range altnames {
-		if len(validation.IsDNS1123Subdomain(altname)) != 0 && net.ParseIP(altname) == nil {
-			allErrs = append(allErrs, field.Invalid(fldPath, altname, "altname is not a valid dns label or ip address"))
+		if errs := validation.IsDNS1123Subdomain(altname); len(errs) != 0 && net.ParseIP(altname) == nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, altname, "altname is not a valid DNS label("+strings.Join(errs, "; ")+") or IP address"))
```

Comment:
`altname is not a valid DNS label("+strings.Join(errs, "; ")+") or IP address")`
this formatting is not ideal.

how about appending `errs` to `allErrs`?
also can you show what errors are combined by strings.Join() as an example?

Reply:
@neolit123 
The errors output example is shown below:
```
[etcd.local.serverCertSANs: Invalid value: "Ubuntu-1804-01": altname is not a valid DNS label(a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')) or IP address]
```

Reply:
@neolit123 
what errors are combined by strings.Join():
```
must be no more than 253 characters; a DNS-1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

Reply:
the `or IP address` at the end is weird in  case of long error.
try placing it in the beginning.

Reply:
also please use Sprintf() instead of the string + string.